src/main/java/com/exasol/parquetio/merger/ChunkIntervalMerger.java

```diff
@@ -0,0 +1,57 @@
+package com.exasol.parquetio.merger;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.parquetio.data.ChunkInterval;
+import com.exasol.parquetio.data.ChunkIntervalImpl;
+
+/**
+ * A class that sorts and merges list of {@link ChunkInterval}s.
+ */
+public class ChunkIntervalMerger {
+
+    /**
+     * Sorts and merges overlapping chunks.
+     *
+     * @param chunks a list {@link ChunkInterval} chunks
+     * @return a sorted and merged chunks
+     */
+    public List<ChunkInterval> sortAndMerge(final List<ChunkInterval> chunks) {
+        if (chunks == null || chunks.isEmpty()) {
+            throw new IllegalArgumentException(
+                    ExaError.messageBuilder("E-PIOJ-5").message("Chunk intervals list is empty.")
+                            .mitigation("Please provide a valid list of Parquet file chunks.").toString());
+        }
```

Comment:
Yes, that would be nice, but here it is required that list is not empty. Otherwise, reading would fail. But maybe it is not concern of this class, I will move it to constructor.